java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,174 @@
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;MapKey of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Element of MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Collection);;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;(String,Object[]);;ArrayElement of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;MapKey of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;scheme;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;scheme;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;userInfo;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;userInfo;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilderFactory;true;builder;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilderFactory;true;uriString;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;UriComponents$UriTemplateVariables;true;getValue;;;MapValue of Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;(String,Map);;MapValue of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
```

Comment:
These are defined on the abstract superclass too

Reply:
I assume you meant the `DefaultUriBuilderFactory` models below; I've removed them.

Reply:
It seems to me like `AbstractUriTemplateHandler` is an implementation of `UriTemplateHandler`?

Reply:
Doh yes sorry, I read the class hierarchy backwards